### ISSUE TITLE
Red lines captured from iOS device are very pixelated

### ISSUE THREAD

**Issue by ybizeul** [obsproject/obs-studio#2960]
## Platform

Operating system and version: macOS 10.15.5
OBS Studio version: 25.0.8

## Expected Behavior
When capturing iPad screen, *red* parts shouldn't be pixelated

## Current Behavior
For some reason, only red strokes or areas are very pixelated.

![Screenshot 2020-05-20 at 22 23 34](https://user-images.githubusercontent.com/182035/82494042-42620900-9ae9-11ea-93b7-6ebe138ec3c4.png)

On the left, "Video Capture Device" (USB attached iPad Pro)
On the right, "Window Capture" showing QuickTime Player window displaying iPad Pro screen

Specifically the red strokes are very pixelated but appears normally with QuickTime Player capture.

## Steps to Reproduce

1. Connect an iPad to a mac
2. Add the iPad "Video Capture Device" to the scene
3. Stroke a red line, using Microsoft Whiteboard on the iPad for example
4. Pixels are clearly visible
5. Stroke a different color line, that has no red
6. Much smoother

##
The only workaround I found was to put the QuickTime Player window in the scene instead of the direct capture of the iPad screen in OBS

**Comment by Fenrirthviti** [comment]
Please provide a log from OBS. It looks like the resolution of the direct capture is just set to something lower than the native resolution of the device, causing scaling quality loss.

**Comment by derrod** [comment]
That's the result of chroma subsampling, and not an OBS issue. See if you can switch the format to something that isn't subsampled or uses more samples (e.g. RGB, YUY2).

For reference see https://en.m.wikipedia.org/wiki/Chroma_subsampling

**Comment by ybizeul** [comment]
![image](https://user-images.githubusercontent.com/182035/82495042-ce286500-9aea-11ea-8e0a-ec3cd9263904.png)
That's using native resolution of the iPad, we still see no antialiasing on the red.

Unfortunately, I don't see any way to change the capture method.

If I un-check "Preset" I get this and cannot put a value that works :
![image](https://user-images.githubusercontent.com/182035/82495419-5eff4080-9aeb-11ea-8ece-5e8693af3b49.png)


Note that removing the color key doesn't change the result :
```
22:37:37.510: CPU Name: Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
22:37:37.510: CPU Speed: 2900MHz
22:37:37.510: Physical Cores: 4, Logical Cores: 8
22:37:37.510: Physical Memory: 16384MB Total
22:37:37.510: OS Name: Mac OS X (NSMACHOperatingSystem)
22:37:37.510: OS Version: Version 10.15.3 (Build 19D76)
22:37:37.510: Kernel Version: 19.3.0
22:37:37.515: hotkeys-cocoa: Getting keyboard keys failed
22:37:37.515: hotkeys-cocoa: Getting keyboard keys failed
22:37:37.515: hotkeys-cocoa: Getting keyboard keys failed
22:37:37.515: hotkeys-cocoa: Using layout 'com.apple.keylayout.French'
22:37:37.516: Portable mode: false
22:37:37.680: OBS 25.0.8 (mac)
22:37:37.680: ---------------------------------
22:37:37.681: ---------------------------------
22:37:37.681: audio settings reset:
22:37:37.681: 	samples per sec: 44100
22:37:37.681: 	speakers:        2
22:37:37.685: ---------------------------------
22:37:37.685: Initializing OpenGL...
22:37:37.720: Loading up OpenGL on adapter ATI Technologies Inc. AMD Radeon Pro 560 OpenGL Engine
22:37:37.720: OpenGL loaded successfully, version 4.1 ATI-3.5.5, shading language 4.10
22:37:37.822: ---------------------------------
22:37:37.822: video settings reset:
22:37:37.822: 	base resolution:   1920x1080
22:37:37.822: 	output resolution: 1280x720
22:37:37.822: 	downscale filter:  Bicubic
22:37:37.822: 	fps:               30/1
22:37:37.822: 	format:            NV12
22:37:37.822: 	YUV mode:          601/Partial
22:37:37.822: NV12 texture support not available
22:37:37.824: Audio monitoring device:
22:37:37.824: 	name: Default
22:37:37.824: 	id: default
22:37:37.824: ---------------------------------
22:37:37.829: Failed to load 'en-US' text for module: 'decklink-ouput-ui.so'
22:37:37.842: os_dlopen(libpython3.7m.dylib->libpython3.7m.dylib): dlopen(libpython3.7m.dylib, 257): image not found
22:37:37.842: 
22:37:37.842: [Python] Could not load library: libpython3.7m.dylib
22:37:37.914: A DeckLink iterator could not be created.  The DeckLink drivers may not be installed
22:37:37.914: No blackmagic support
22:37:38.073: [VideoToolbox encoder]: Adding VideoToolbox H264 encoders
22:37:38.187: [obs-browser]: Version 2.8.6
22:37:38.241: VLC found, VLC video source enabled
22:37:38.256: Failed to load 'en-US' text for module: 'obs-mac-virtualcam.so'
22:37:38.257: ---------------------------------
22:37:38.257:   Loaded Modules:
22:37:38.257:     obs-mac-virtualcam.so
22:37:38.257:     logi_obs_plugin.so
22:37:38.257:     vlc-video.so
22:37:38.257:     text-freetype2.so
22:37:38.257:     rtmp-services.so
22:37:38.257:     obs-x264.so
22:37:38.257:     obs-vst.so
22:37:38.257:     obs-transitions.so
22:37:38.257:     obs-outputs.so
22:37:38.257:     obs-libfdk.so
22:37:38.257:     obs-filters.so
22:37:38.257:     obs-ffmpeg.so
22:37:38.257:     obs-browser.so
22:37:38.257:     mac-vth264.so
22:37:38.257:     mac-syphon.so
22:37:38.257:     mac-decklink.so
22:37:38.257:     mac-capture.so
22:37:38.257:     mac-avcapture.so
22:37:38.257:     linux-jack.so
22:37:38.257:     image-source.so
22:37:38.257:     frontend-tools.so
22:37:38.257:     decklink-ouput-ui.so
22:37:38.257:     coreaudio-encoder.so
22:37:38.257: ---------------------------------
22:37:38.276: os_dlopen(../obs-plugins/obs-browser->../obs-plugins/obs-browser.so): dlopen(../obs-plugins/obs-browser.so, 257): image not found
22:37:38.276: 
22:37:38.276: ==== Startup complete ===============================================
22:37:38.325: Service '' not found
22:37:38.330: All scene data cleared
22:37:38.330: ------------------------------------------------
22:37:38.424: coreaudio: device 'Soundflower (2ch)' initialized
22:37:38.424: [Loaded global audio device]: 'Desktop Audio'
22:37:38.428: coreaudio: device 'Blue Snowball' initialized
22:37:38.428: [Loaded global audio device]: 'Mic/Aux'
22:37:38.460: Camera: Selected device 'C922 Pro Stream Webcam'
22:37:38.464: Camera: Using preset 1280x720
22:37:38.601: adding 46 milliseconds of audio buffering, total audio buffering is now 46 milliseconds (source: Mic/Aux)
22:37:38.601: 
22:37:38.808: iPad Horizontal: Selected device 'IPad Pro'
22:37:38.808: iPad Horizontal: Using preset 1280x720
22:37:38.843: Switched to scene 'Scene'
22:37:38.847: ------------------------------------------------
22:37:38.847: Loaded scenes:
22:37:38.847: - scene 'Scene':
22:37:38.857:     - source: 'Fond Bleu NetApp' (image_source)
22:37:38.857:     - source: 'Camera' (av_capture_input)
22:37:38.857:         - filter: 'Chroma Key' (chroma_key_filter)
22:37:38.857:         - filter: 'Chroma Key 2' (chroma_key_filter)
22:37:38.857:     - source: 'iPad Horizontal' (av_capture_input)
22:37:38.857:         - filter: 'Color Key' (color_key_filter)
22:37:38.857:     - source: 'Window Capture' (window_capture)
22:37:38.857: ------------------------------------------------
22:38:13.336: iPad Horizontal: Selected preset AVCaptureSessionPresetHigh
```

**Comment by ybizeul** [comment]
Same thing if I set 1920x1080 :

```
22:51:47.230: 	base resolution:   1920x1080
22:51:47.230: 	output resolution: 1920x1080
```

**Comment by gxalpha** [comment]
This is the result of preset being set to a resolution. It needs to be set to "High" (which usually defaults to the max resolution of the device), anything else is too low of a resolution.
QuickTime Player only has "High" (as well as "Maximum" which from what I can tell does the same in most cases) available, which is why the problem doesn't appear there.

**Comment by RytoEX** [comment]
We believe this was resolved by #6634. If this is still an issue in the latest CI builds, please feel free to reopen this Issue.
